config/secretsmanager/config.go

```diff
@@ -13,5 +22,193 @@ func Configure(p *config.Provider) {
 		r.TerraformConfigurationInjector = func(_ map[string]any, params map[string]any) {
 			params["name_prefix"] = ""
 		}
+		r.TerraformCustomDiff = func(diff *terraform.InstanceDiff, state *terraform.InstanceState, config *terraform.ResourceConfig) (*terraform.InstanceDiff, error) {
+			// skip diff customization on create
+			if state == nil || state.Empty() {
+				return diff, nil
+			}
+			if config == nil {
+				return nil, errors.New("resource config cannot be nil")
+			}
+			// skip no diff or destroy diffs
+			if diff == nil || diff.Empty() || diff.Destroy || diff.Attributes == nil {
+				return diff, nil
+			}
+
+			currentReplicaSet, ok := r.TerraformResource.Data(state).Get("replica").(*schema.Set)
+			if !ok {
+				return nil, errors.New("could not read \"replica\" from state")
+			}
+			nisReplica, ok := config.Get("replica")
+			if !ok {
+				return nil, errors.New("could not read replica block from config")
+			}
+			desiredReplicaList := nisReplica.([]interface{})
+			// this is the hash implementation of *schema.Set, which is unexported
+			hashFunc := func(val interface{}) string {
+				code := currentReplicaSet.F(val)
+				if code < 0 {
+					code = -code
+				}
+				return strconv.Itoa(code)
+			}
+			// the number of
+			desiredReplicasWithRegionOnly := make(map[string]int)
+			//
+			// map[region]map[kms_key_id]hash
+			regionsInCurrentState := make(map[string]map[string]string)
+			type replica struct {
+				KMSKeyID string `json:"kms_key_id"`
+				Region   string `json:"region"`
+			}
+
+			// traverse state
+			for _, v := range currentReplicaSet.List() {
+				// v is an interface{} type, which is a replica
+				// marshal then unmarshal to convert it to the internal type replica for easier access to fields
+				replicaBytes, err := json.JSParser.Marshal(v)
+				if err != nil {
+					continue
+				}
+				cReplica := &replica{}
+				if err := json.JSParser.Unmarshal(replicaBytes, cReplica); err != nil {
+					return nil, err
+				}
+				if cReplica.Region == "" || cReplica.KMSKeyID == "" {
+					// we should not be here, replicas at current state always have their region set and kms_key_id computed
+					continue
```

Comment:
It seems like we should do _something_ if we somehow get here, at least logging a warning.